client/my-sites/email/email-management/titan-management-nav/index.jsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { CompactCard } from '@automattic/components';
+import {
+	emailManagementManageTitanAccount,
+	emailManagementNewTitanAccount,
+} from 'calypso/my-sites/email/paths';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { fetchTitanAutoLoginURL } from 'calypso/my-sites/email/email-management/titan-functions';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getManagePurchaseUrlFor } from 'calypso/my-sites/purchases/paths';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import { getTitanMailOrderId } from 'calypso/lib/titan/get-titan-mail-order-id';
+import { isEnabled } from 'calypso/config';
+import SectionHeader from 'calypso/components/section-header';
+import VerticalNav from 'calypso/components/vertical-nav';
+import VerticalNavItem from 'calypso/components/vertical-nav/item';
+
+class TitanManagementNav extends React.Component {
+	static propTypes = {
+		domain: PropTypes.object.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			isFetchingExternalManagementUrl: this.shouldFetchControlPanelLink(),
+			externalManagementUrl: '',
+		};
+	}
+
+	shouldFetchControlPanelLink() {
+		return ! isEnabled( 'titan/iframe-control-panel' );
+	}
+
+	componentDidMount() {
+		this._mounted = true;
+
+		if ( this.shouldFetchControlPanelLink() ) {
+			const { domain, translate } = this.props;
+
+			fetchTitanAutoLoginURL( getTitanMailOrderId( domain ) ).then( ( { error, loginURL } ) => {
+				if ( this._mounted ) {
+					this.setState( { isFetchingExternalManagementUrl: false } );
+				}
+				if ( error ) {
+					this.props.errorNotice(
+						error ?? translate( 'An unknown error occurred. Please try again later.' )
+					);
+				} else if ( this._mounted ) {
+					this.setState( { externalManagementUrl: loginURL } );
+				}
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		this._mounted = false;
+	}
+
+	renderTitanManagementLink = () => {
+		const { currentRoute, domain, selectedSiteSlug, translate } = this.props;
+		const linkTitle = translate( 'Manage your email settings and accounts' );
+		if ( isEnabled( 'titan/iframe-control-panel' ) ) {
+			return (
+				<VerticalNavItem
+					path={ emailManagementManageTitanAccount( selectedSiteSlug, domain.name, currentRoute ) }
+				>
+					{ linkTitle }
+				</VerticalNavItem>
+			);
+		}
+		const { externalManagementUrl, isFetchingExternalManagementUrl } = this.state;
+		if ( ! externalManagementUrl || isFetchingExternalManagementUrl ) {
+			return <VerticalNavItem isPlaceholder={ true } />;
+		}
+		return <VerticalNavItem path={ externalManagementUrl }>{ linkTitle }</VerticalNavItem>;
```

Comment:
We may want to have the `external` attribute set to true here so the item clearly indicates it is an external link. (Though it then opens in a new window, which is a change in behaviour from what we have now.)